admin_guide/encrypting_data.adoc

```diff
@@ -0,0 +1,291 @@
+[[admin-guide-encrypting-data-at-datastore]]
+= Encrypting Data at Datastore Layer
+:data-uri:
+:icons:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+[[encrypting-data-overview]]
+== Overview
+
+This topic shows how to enable and configure encryption of secret data at datastore layer. While
+examples use the `secrets` resource, any resource can be encrypted. Yet another candidate is
+`configmaps`.
+
+[[encrypting-data-configuration]]
+== Configuration and Determining Whether Encryption Is Already Enabled
+
+In order to activate experimental support for data encryption, an argument
+*--experimental-encryption-provider-config* should be passed to the Kubernetes API server:
+
+.Excerpt of *_master-config.yaml_*
+====
+
+[source,yaml]
+----
+kubernetesMasterConfig:
+  apiServerArguments:
+    experimental-encryption-provider-config:
+    - /path/to/encryption-config.yaml
+----
+====
+
+For more information about *_master-config.yaml_* and its format, see
+xref:install_config/master_node_configuration.html#master-configuration-files[Master Configuration
+Files] topic.
+
+[[encrypting-data-encryption-config]]
+== Understanding the Encryption Configuration
+
+.Encryption configuration file with all available providers
+====
+
+[source,yaml]
+----
+kind: EncryptionConfig
+apiVersion: v1
+resources: <1>
+  - resources: <2>
+    - secrets
+    providers: <3>
+    - aescbc: <4>
+        keys:
+        - name: key1 <5>
+          secret: c2VjcmV0IGlzIHNlY3VyZQ== <6>
+        - name: key2
+          secret: dGhpcyBpcyBwYXNzd29yZA==
+    - secretbox:
+        keys:
+        - name: key1
+          secret: YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXoxMjM0NTY=
+    - aesgcm:
+        keys:
+        - name: key1
+          secret: c2VjcmV0IGlzIHNlY3VyZQ==
+        - name: key2
+          secret: dGhpcyBpcyBwYXNzd29yZA==
+    - identity: {}
+----
+<1> Each `resources` array item is a separate config and contains a complete configuration.
+<2> The `resources.resources` field is an array of Kubernetes resource names (*resource* or
+*resource.group*) that should be encrypted.
+<3> The `providers` array is an ordered xref:encrypting-data-providers[list of
+the possible encryption providers]. Only one provider type may be specified per entry (*identity* or
+*aescbc* may be provided, but not both in the same item).
```

Comment:
IMHO if they decided to encrypt the data they probably already know what it is or they don't even need to know the details and just follow our recommendations.